scripts/zones/Behemoths_Dominion/mobs/King_Behemoth.lua

```diff
@@ -15,6 +15,18 @@ function onMobInitialize(mob)
     mob:setMobMod(MOBMOD_MAGIC_COOL, 60);
 end;
 
+-----------------------------------
+-- onAdditionalEffect Action
+-----------------------------------
+function onAdditionalEffect(mob,target,damage)
+    if ((math.random(1,100) >= 81) or (target:hasStatusEffect(EFFECT_STUN) == true)) then

```

Comment:
Update and new comment post so your aware: 

1. After posting I remembered that the existing trunk examples for mobs with melee **status effects** very much suck. I don't think even one is retail accurate. The element dmg and drain ones are alright but the status ones look to be just random like you were doing. So I for one am not gonna expect perfection here - **if a core dev doesn't come along and insist on a resist check, I'm not going to either, and it can stay just a fully random thing till how those should be done is sorted out.**

Also:

2. I maybe should have pointed you to one of the other resist functions (there some in magic.lua and at least 1 in monstertpmove.lua as well) instead of `applyResistanceAddEffect(mob,target,ELE_LIGHTNING, EFFECT_STUN)`

3. I just looked and that function for the resist I pointed you at hasn't been updated yet, but thats what @maxtherabbit said to do (tag, you're it!) for the status effects in addEffect scripts when we were doing items. No idea if mobs should use same thing or not and I think I used more then 1 resist function on those items anyways...  >.>